model/common/src/icon4py/model/common/grid/horizontal.py

```diff
@@ -46,199 +46,135 @@
-_MIN_RL_EDGE: Final[int] = _MIN_RL_EDGE_INT - (2 * NUM_GHOST_ROWS + 1)
-_MAX_RL_EDGE: Final[int] = 2 * _MAX_RL_CELL
-
-_LATERAL_BOUNDARY_EDGES: Final[int] = 1 + _ICON_INDEX_OFFSET_EDGES  # 14
-_INTERIOR_EDGES: Final[int] = _ICON_INDEX_OFFSET_EDGES  # 13
-_NUDGING_EDGES: Final[int] = _GRF_BOUNDARY_WIDTH_EDGES + _ICON_INDEX_OFFSET_EDGES  # 22
-_HALO_EDGES: Final[int] = _MIN_RL_EDGE_INT - 1 + _ICON_INDEX_OFFSET_EDGES  # 4
-_LOCAL_EDGES: Final[int] = _MIN_RL_EDGE_INT + _ICON_INDEX_OFFSET_EDGES  # 5
-_END_EDGES: Final[int] = 0
-
-_LATERAL_BOUNDARY_CELLS: Final[int] = 1 + _ICON_INDEX_OFFSET_CELLS  # 9
-_INTERIOR_CELLS: Final[int] = _ICON_INDEX_OFFSET_CELLS  # 8
-_NUDGING_CELLS: Final[int] = _GRF_BOUNDARY_WIDTH_CELL + 1 + _ICON_INDEX_OFFSET_CELLS  # 13
-_HALO_CELLS: Final[int] = _MIN_RL_CELL_INT - 1 + _ICON_INDEX_OFFSET_CELLS  # 3
-_LOCAL_CELLS: Final[int] = _MIN_RL_CELL_INT + _ICON_INDEX_OFFSET_CELLS  # 4
-_END_CELLS: Final[int] = 0
-
-_LATERAL_BOUNDARY_VERTICES = 1 + _ICON_INDEX_OFFSET_VERTEX  # 8
-_INTERIOR_VERTICES: Final[int] = _ICON_INDEX_OFFSET_VERTEX  # 7
-_NUDGING_VERTICES: Final[int] = 0
-_HALO_VERTICES: Final[int] = _MIN_RL_VERTEX_INT - 1 + _ICON_INDEX_OFFSET_VERTEX  # 2
-_LOCAL_VERTICES: Final[int] = _MIN_RL_VERTEX_INT + _ICON_INDEX_OFFSET_VERTEX  # 3
-_END_VERTICES: Final[int] = 0
-
-
 _EDGE_GRF: Final[int] = 24
 _CELL_GRF: Final[int] = 14
 _VERTEX_GRF: Final[int] = 13
 
-
-_LATERAL_BOUNDARY = {
-    dims.CellDim: _LATERAL_BOUNDARY_CELLS,
-    dims.EdgeDim: _LATERAL_BOUNDARY_EDGES,
-    dims.VertexDim: _LATERAL_BOUNDARY_VERTICES,
+_ICON_LATERAL_BOUNDARY: Final[dict[gtx.Dimension, int]] = {
+    dims.CellDim: 9,
+    dims.EdgeDim: 14,
+    dims.VertexDim: 8,
 }
-_LOCAL = {
-    dims.CellDim: _LOCAL_CELLS,
-    dims.EdgeDim: _LOCAL_EDGES,
-    dims.VertexDim: _LOCAL_VERTICES,
+ICON_LOCAL = {
+    dims.CellDim: 4,
+    dims.EdgeDim: 5,
+    dims.VertexDim: 3,
 }
-_HALO = {
-    dims.CellDim: _HALO_CELLS,
-    dims.EdgeDim: _HALO_EDGES,
-    dims.VertexDim: _HALO_VERTICES,
+_ICON_HALO = {
+    dims.CellDim: 3,
+    dims.EdgeDim: 4,
+    dims.VertexDim: 2,
 }
-_INTERIOR = {
-    dims.CellDim: _INTERIOR_CELLS,
-    dims.EdgeDim: _INTERIOR_EDGES,
-    dims.VertexDim: _INTERIOR_VERTICES,
+_ICON_INTERIOR = {
+    dims.CellDim: 8,
+    dims.EdgeDim: 13,
+    dims.VertexDim: 7,
 }
-_NUDGING = {
-    dims.CellDim: _NUDGING_CELLS,
-    dims.EdgeDim: _NUDGING_EDGES,
-    dims.VertexDim: _NUDGING_VERTICES,
+_ICON_NUDGING = {
+    dims.CellDim: 13,
+    dims.EdgeDim: 22,
+    dims.VertexDim: 12,
 }
-_END = {
-    dims.CellDim: _END_CELLS,
-    dims.EdgeDim: _END_EDGES,
-    dims.VertexDim: _END_VERTICES,
+_ICON_END = {
+    dims.CellDim: 0,
+    dims.EdgeDim: 0,
+    dims.VertexDim: 0,
 }
 
-_BOUNDS = {
+_ICON_CONSTANTS_BOUNDS = {
     dims.CellDim: (0, _CELL_GRF - 1),
     dims.EdgeDim: (0, _EDGE_GRF - 1),
     dims.VertexDim: (0, _VERTEX_GRF - 1),
 }
+"""
+Indices used to look up start index and end index in the arrays int `start_idx_[c,e,v]` from ICON.
```

Comment:
Is this docstring for all or only the last?

Reply:
For all